### PR TITLE
feat(release): allow using existing draft release

### PR DESCRIPTION
**Pull request by mloberg**
The PR allows setting a `release.use_existing_draft` config option to use an existing draft release instead of creating a new release.

Our release process creates a draft PR before calling GoReleaser. While we can delete an old draft and create a new one, it would be nice to keep the existing draft PR and use that instead of creating a new one.

### CODE REVIEW

**Comment by mloberg** [comment]
Sorry to tag you @caarlos0, but I was hoping to get some traction on this change. I'm open to feedback on it and if we wanted to change any of the config options for it. Thanks!

**Comment by caarlos0** [comment]
hey, sorry for the delay! somehow missed the notification

can you add tests for it as well?

File: www/docs/customization/release.md [hunk 1]
```diff
@@ -37,6 +37,11 @@ release:
   # Available only for GitHub.
   replace_existing_draft: true
 
+  # Whether to use an existing draft release as the target release.
+  #
+  # Available only for GitHub.
```
Comment (caarlos0) [inline www/docs/customization/release.md:42]:
```suggestion
  # Available only for GitHub.
  # Since: v2.5.
```

**Comment by mloberg** [comment]
> can you add tests for it as well?

There's no existing tests for the [release section](https://github.com/goreleaser/goreleaser/blob/f54fb113ee6415fd2ada737b3ffd71e5e32682ad/internal/client/github_test.go#L996-L999). I can work on adding these if that's needed to contribute this change

**Comment by caarlos0** [comment]
oh, right!

can you give me a few days? will try to fix that TODO, and then we can add tests for this as well. 🤔

**Comment by mloberg** [comment]
> oh, right!
> 
> can you give me a few days? will try to fix that TODO, and then we can add tests for this as well. 🤔

No problem. I'm also happy to help if you want

**Comment by mloberg** [comment]
Following up on this. Have you been able to add any tests? Do you want any help with that?

**Comment by caarlos0** [comment]
> Following up on this. Have you been able to add any tests? Do you want any help with that?

sorry, got side tracked, i can probably still take a look before releasing 2.5. If you feel like doing it, though, feel free to as well!

**Comment by mloberg** [comment]
> > Following up on this. Have you been able to add any tests? Do you want any help with that?
> 
> sorry, got side tracked, i can probably still take a look before releasing 2.5. If you feel like doing it, though, feel free to as well!

No worries. I'll see what time I can dedicate to it this week. Thanks!

**Comment by mloberg** [comment]
@caarlos0 I added one test for create release. Please let me know how that's looking and if there's any stylistic changes I should make before adding more tests. Thanks!

**Review by caarlos0 — APPROVED** [review]

**Comment by caarlos0** [comment]
LGTM, thank you so much @mloberg!

**Comment by mloberg** [comment]
Alright. I think I've got the tests needed for this feature in place. Let me know if anything is missing or needs to be updated. Thanks!

**Review by caarlos0 — APPROVED** [review]
this looks great!

thank you so much 💜

**Pull request merged**